### ISSUE TITLE
[Bug] Torch divide_no_nan() produces NaN gradients

### ISSUE THREAD

**Issue by ChiragSW** [keras-team/keras#23063]
The Torch implementation uses ```torch.where(x2 == 0, 0, torch.divide(x1, x2))```. Torch still evaluates the divide branch, so forward values can be zero while backward gradients become NaN for zero denominators. JAX and TensorFlow donot do this.

Repro Script: 
```python
import os
os.environ["KERAS_BACKEND"] = "torch"

import keras
import torch

x1 = torch.tensor([1.0, 2.0], requires_grad=True)
x2 = torch.tensor([0.0, 2.0], requires_grad=True)

y = keras.ops.divide_no_nan(x1, x2)
print("y:", y)

y.sum().backward()
print("x1.grad:", x1.grad)
print("x2.grad:", x2.grad)
print("finite:", torch.isfinite(x1.grad), torch.isfinite(x2.grad))
```

output:
```
y: tensor([0., 1.], grad_fn=<WhereBackward0>)
x1.grad: tensor([   nan, 0.5000])
x2.grad: tensor([    nan, -0.5000])
finite: tensor([False,  True]) tensor([False,  True])
```
Existing Keras tests only check forward values, so they miss this gradient bug.

Enviroment: keras 3.15.0
